src/syntax/syntax.rkt

```diff
@@ -24,56 +27,97 @@
   [bf (->* () bigvalue?)]
   [fl (->* () value?)]
   [ival (or/c (->* () ival?) #f)]
-  [nonffi (->* () value?)]
-  [->tex string?])
+  [nonffi (->* () value?)])
 
 (define (constant-info constant field) (table-ref constants constant field))
 
 (define-syntax-rule (define-constant constant ctype [key value] ...)
   (table-set! constants 'constant
               (make-hash (list (cons 'type 'ctype) (cons 'key value) ...))))
 
-(define-constant PI real
+(define parametric-constants (make-hash))
+(define parametric-constants-reverse (make-hash))
+
+(define (get-parametric-constant name type)
+  (for/or ([sig (hash-ref parametric-constants name)])
+    (match-define (list* true-name rtype) sig)
+    (and
+     (equal? rtype type)
+     true-name)))
+
+(define (declare-parametric-constant! name cnst type)
+  (hash-update! parametric-constants name (curry cons (list* cnst type)) '())
+  (hash-set! parametric-constants-reverse cnst name))
+
+(define-constant PI.f64 binary64
   [bf (λ () pi.bf)]
   [fl (λ () pi)]
   [ival ival-pi]
-  [nonffi (λ () pi)]
-  [->tex "\\pi"])
+  [nonffi (λ () pi)])
 
-(define-constant E real
+(define-constant E.f64 binary64
   [bf (λ () (bfexp 1.bf))]
   [fl (λ () (exp 1.0))]
   [ival ival-e]
-  [nonffi (λ () (exp 1.0))]
-  [->tex "e"])
+  [nonffi (λ () (exp 1.0))])
```

Comment:
Now that I think of it I'm not sure constants need `nonffi` versions. In fact, it might make sense to revisit how `nonffi` is structured. Trello'd.